doc/external.md

```diff
@@ -0,0 +1,68 @@
+**WORK IN PROGRESS, NOT YET COMPLETE**
+
+# External (template)
+
+The External template class implements the ability to create a Value object with arbitrary C++ data. It is the user's responsibility to manage the memory for the arbitrary C++ data.
+
+External objects can be created with an optional Finalizer function and optional Hint value. The Finalizer function, if specified, is called when your External object is released by Node's garbage collector. It gives your code the opportunity to free any dynamically created data. If you specify a Hint value, it is passed to your Finalizer function.
+
+## Methods
+
+### New
+
+```cpp
+template <typename T>
+static External New(napi_env env, T* data);
+```
+
+- `[in] env`: The `napi_env` environment in which to construct the External object.
+- `[in] data`: The arbitrary C++ data to be held by the External object.
+
+Returns the created `External<T>` object.
+
+### New
+
+```cpp
+template <typename T>
+static External New(napi_env env,
+			T* data,
+			Finalizer finalizeCallback);
+```
+
+- `[in] env`: The `napi_env` environment in which to construct the External object.
+- `[in] data`: The arbitrary C++ data to be held by the External object.
+- `[in] finalizeCallback`: A function called when the External object is released by the garbage collector accepting a T* and returning void.
+
+Returns the created `External<T>` object.
+
+### New
+
+```cpp
+template <typename T>
+static External New(napi_env env,
+			T* data,
+			Finalizer finalizeCallback,
+			Hint* finalizeHint);
+```
+
+- `[in] env`: The `napi_env` environment in which to construct the External object.
+- `[in] data`: The arbitrary C++ data to be held by the External object.
+- `[in] finalizeCallback`: A function called when the External object is released by the garbage collector accepting T* and Hint* parameters and returning void.
+- `[in] finalizeHint`: A hint value passed to the `finalizeCallback` function.
+
+Returns the created `External<T>` object.
+
+### Data
+
+```cpp
+T* Data() const;
+```
+
+Returns a pointer to the arbitrary C++ data held by the External object.
+
+### Not documented here
+
```

Comment:
Is this just because the doc will be added in future PR ?  Or do you have some questions that need to be answered before you can add the doc for these ?

Reply:
I wanted the reviewers to know that they're not documented. I'm pretty sure there's no need to document them. If so, I'll remove the note.

Reply:
I'm thinking best to remove.  It confused me so it might confuse others as well.